### ISSUE TITLE
SF Next Muni is totally broken

### ISSUE THREAD

**Issue by sloanesturz** [tidbyt/community#804]
My most favorite applet, the SF Next Muni is totally broken. I think that the API changed out from underneath it?

If you check the API's list of "agencies", "sfmuni" is no longer present. There is an entry called "sfmuni-sandbox" [here](https://retro.umoiq.com/service/publicJSONFeed?command=agencyList) but it doesn't appear to have any valid predictions when you check for a specific location [example](https://retro.umoiq.com/service/publicJSONFeed?command=predictions&a=sfmuni-sandbox&stopId=13909&useShortTitles=true&r=N)

It seems like this applet needs to migrate to a new API. Potentially we can use the data directly from the 511 government agency? https://511.org/open-data/transit

@juniormonkey @dja852, curious if you have any ideas or considered using the 511 API in the initial implementation of the app. I am happy to work on the migration but would prefer not to chasing loose ends if you know more context behind this breakage.

**Comment by juniormonkey** [comment]
I don't know what's going on with the muni API. Looks like they're doing some system upgrades this weekend: https://www.sfmta.com/blog/sfmta-upgrades-predictions-software-weekend

I'll wait for the dust to settle tomorrow morning and see if the nextbus API recovers its sfmuni agency; if not, I'll see if 511's "real-time stop monitoring" API can replace it.

**Comment by sloanesturz** [comment]
Thanks for the pointer to the SFMTA update. Let's hope that's the issue and it bounces back on its own.

**Comment by dja852** [comment]
I've been in contact with them. Although some data is being fed back to NextBus, it is not being made available in that API. I think 511/MTC is the source that needs to be used now, but I'm waiting for a follow up

**Comment by juniormonkey** [comment]
#807 fixes predictions. I have a plan for messages but it's a little more involved since GTFS targets messages by station, line, trip and operator rather than just assigning high/med/low priority, so I'll revamp the filtering options in the app as I update the API requests. But for now, at least arrival times should be showing up again.

**Comment by dja852** [comment]
@juniormonkey I've heard back from SFMTA that there is a new umoIQ API that you can request access to. Interested? Or do you want to stick with 511 API for now?

**Comment by juniormonkey** [comment]
@dja852 thanks - I'm curious, but the work to use 511 API is done already, so *shrug*. I don't prefer one over the other, unless one is more likely to be more accurate and/or reliable? Have you heard anything of that from SFMTA?

**Comment by dja852** [comment]
@juniormonkey They are supposed to be pulling from the same data source. If there is any lag it shouldn't be noticable... i'll do some comparisons once your updates are re-released (as I still don't see the app available)

**Comment by juniormonkey** [comment]
I see the app in the directory (albeit without data this morning) - is it
not there for you?

The problem it's been running into is hitting it's request quota. If the
UmoIQ API has a higher limit (or no limit at all like the previous one
did), we should definitely migrate since even slightly laggy data is better
than none at all.

**Comment by juniormonkey** [comment]
Could you point me to the new UmoIQ API so I can investigate that? Thanks!

**Comment by dja852** [comment]
@juniormonkey Sorry for late reply .. I added you here with details I received on this. Not public right now because not sure how public it is meant to be https://github.com/dja852/umoiq

**Comment by juniormonkey** [comment]
Fabulous, thanks - I'll investigate this.

**Comment by juniormonkey** [comment]
Current status:

I've reached out to Cubic to request UmoIQ API access; I'm waiting to receive their TOS and pricing schedule, and then apparently they also need permission from SFMTA.

In parallel, I requested a quota increase from 511.org; they replied that they are unable to accommodate an increased API rate.

At this point, I fear the only solution is to require applet users to register their own API keys and enter them in a new config setting.
